pkgs/tools/misc/coreutils/default.nix

```diff
@@ -75,80 +84,89 @@ stdenv.mkDerivation (rec {
     ];
 
-
   buildInputs = [ gmp ]
-    ++ optional aclSupport acl
-    ++ optional attrSupport attr
-    ++ optional withOpenssl openssl
-    ++ optionals selinuxSupport [ libselinux libsepol ]
-       # TODO(@Ericson2314): Investigate whether Darwin could benefit too
-    ++ optional (stdenv.hostPlatform != stdenv.buildPlatform && stdenv.hostPlatform.libc != "glibc") libiconv;
+    ++ lib.optional aclSupport acl
+    ++ lib.optional attrSupport attr
+    ++ lib.optional withOpenssl openssl
+    ++ lib.optionals selinuxSupport [ libselinux libsepol ]
+    # TODO(@Ericson2314): Investigate whether Darwin could benefit too
+    ++ lib.optional (isCross && stdenv.hostPlatform.libc != "glibc") libiconv;
 
   # The tests are known broken on Cygwin
   # (http://article.gmane.org/gmane.comp.gnu.core-utils.bugs/19025),
   # Darwin (http://article.gmane.org/gmane.comp.gnu.core-utils.bugs/19351),
   # and {Open,Free}BSD.
   # With non-standard storeDir: https://github.com/NixOS/nix/issues/512
   # On aarch64+musl, test-init.sh fails due to a segfault in diff.
-  doCheck = stdenv.hostPlatform == stdenv.buildPlatform
+  doCheck = (!isCross)
     && (stdenv.hostPlatform.libc == "glibc" || stdenv.hostPlatform.isMusl)
     && !(stdenv.hostPlatform.isMusl && stdenv.hostPlatform.isAarch64)
     && !stdenv.isAarch32;
 
   # Prevents attempts of running 'help2man' on cross-built binaries.
-  PERL = if stdenv.hostPlatform == stdenv.buildPlatform then null else "missing";
+  PERL = if isCross then "missing" else null;
 
   enableParallelBuilding = true;
 
-  NIX_LDFLAGS = optionalString selinuxSupport "-lsepol";
-  FORCE_UNSAFE_CONFIGURE = optionalString stdenv.hostPlatform.isSunOS "1";
+  NIX_LDFLAGS = lib.optionalString selinuxSupport "-lsepol";
+  FORCE_UNSAFE_CONFIGURE = lib.optionalString stdenv.hostPlatform.isSunOS "1";
   NIX_CFLAGS_COMPILE = []
     # Work around a bogus warning in conjunction with musl.
-    ++ optional stdenv.hostPlatform.isMusl "-Wno-error"
-    ++ optional stdenv.hostPlatform.isAndroid "-D__USE_FORTIFY_LEVEL=0";
+    ++ lib.optional stdenv.hostPlatform.isMusl "-Wno-error"
+    ++ lib.optional stdenv.hostPlatform.isAndroid "-D__USE_FORTIFY_LEVEL=0";
 
   # Works around a bug with 8.26:
   # Makefile:3440: *** Recursive variable 'INSTALL' references itself (eventually).  Stop.
-  preInstall = optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
+  preInstall = lib.optionalString isCross ''
     sed -i Makefile -e 's|^INSTALL =.*|INSTALL = ${buildPackages.coreutils}/bin/install -c|'
   '';
 
-  postInstall = optionalString (stdenv.hostPlatform != stdenv.buildPlatform && !minimal) ''
+  postInstall = lib.optionalString (isCross && !minimal) ''
     rm $out/share/man/man1/*
     cp ${buildPackages.coreutils-full}/share/man/man1/* $out/share/man/man1
   ''
   # du: 8.7 M locale + 0.4 M man pages
-  + optionalString minimal ''
+  + lib.optionalString minimal ''
     rm -r "$out/share"
   '';
 
-  meta = {
+  meta = with lib; {
```

Comment:
This is a more typical usage, because it removes the redundancy of `platforms` and `maintainers`.
On the other hand, bringing `lib` into the scope of all expression is discouraged.

Reply:
This is the most commonly used form.